src/nvim/os/env.c

```diff
@@ -243,19 +243,24 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
         // Verify that we have found the end of a UNIX ${VAR} style variable
         if (src[1] == '{' && *tail != '}') {
           var = NULL;
-        } else if (src[1] == '{') {
-          ++tail;
-        }
+        } else {
+          if (src[1] == '{') {
+            ++tail;
+          }
 #elif defined(MSWIN)
```

Comment:
@pbhandari When was the Windows-specifc control removed? (In the related PR, or the original fork?) I don't know exactly what you're referring to, but I'd say "yes", and then we can discuss it here.